eth-connector-tests/src/utils.rs

```diff
@@ -282,6 +282,43 @@ impl TestContract {
 
         Ok(AccountId::try_from(user.to_string())?)
     }
+
+    pub async fn set_and_check_access_right(&self, acc: &AccountId) -> anyhow::Result<()> {
+        let res = self
+            .contract
+            .call("set_access_right")
+            .args_json((acc,))
+            .gas(DEFAULT_GAS)
+            .transact()
+            .await?;
+        if res.is_failure() {
+            anyhow::bail!("set_access_right failed");
+        }
+
+        let res = self
+            .contract
+            .call("get_access_right")
+            .view()
+            .await?
+            .json::<AccountId>()?;
+        if &res != acc {
+            anyhow::bail!("check access_right fail: {:?} != {:?}", &res, acc);
+        }
+        Ok(())
+    }
+
+    pub async fn set_engine_account(&self, acc: &AccountId) -> anyhow::Result<()> {
+        let res = self
+            .contract
+            .call("set_engine_account")
+            .args_json((&acc,))
+            .gas(DEFAULT_GAS)
+            .transact()
+            .await
+            .unwrap();
+        assert!(res.is_success());
```

Comment:
Does it really make sense to panic there rather than just return `Err`?

Reply:
As I said before, workspace.rs if something returns an error - it is impossible to detect the place where it is.
`unwrap` do that thing.